### ISSUE TITLE
Improving the iOS first-time user experience (FTUE)

### ISSUE THREAD

**Issue by MadLittleMods** [matrix-org/matrix.to#263]
Improving the iOS first-time user experience (FTUE).

### The `matrix.to` **Continue** button should be visible even after clicking it

Tracked by https://github.com/matrix-org/matrix.to/issues/255

This is important to allow users to retry the permalink after they go away and install the native iOS app from the badge or out of band. Or just pressing back in their browser after being redirected to Element Web.

### Block mobile Element web access with the `/mobile_guide`

Tracked by https://github.com/vector-im/element-web/issues/15035

### Improve the `/mobile_guide`

 1. Remove the "Configure your app" section -> https://github.com/vector-im/element-web/issues/21761
 1. Add a [Smart App Banner](https://developer.apple.com/documentation/webkit/promoting_apps_with_smart_app_banners) to suggest the iOS app to people who don't have it downloaded yet
    1. For people who already have the native iOS app, there is already an app banner in mobile Safari to ["Open in the Element app"](https://user-images.githubusercontent.com/558581/162851685-c99ec62a-f31e-4b37-a417-27fb104982a4.png)
 1. Someone even suggested linking to Hydrogen as an alternative -> https://github.com/vector-im/element-web/issues/19870
 
 
 ### Improve the main mobile web views you would see flowing from `matrix.to` links
 
 Tracked by https://github.com/vector-im/element-web/issues/14962
 
Separately, it would be good  to improve the main mobile web views you would see flowing from `matrix.to` links. Doesn't have to be perfect but less squishy would be good. Forcing people to install native mobile apps in order to see content is a bit evil. The good thing is that our mobile guide is dismissible.

 - Room
 - Spam
 - Viewing a message permalink

**Comment by daniellekirkwood** [comment]
As part of the Account Resurrection cycle in the Delight team, we are picking up this issue with the following decisions/ discussion points: 
1. Yes
2. Not right now, we will address the page design in general and lower the importance of the "continue to desktop site" with visual changes.
3.1 Yes
3.2 Done
3.3 Not right now - interesting idea though
3.4 Done
4. Yes, but a quick fix only

👀  @gsouquet

**Comment by t3chguy** [comment]
4. unfortunately wouldn't be a quick fix beyond for rooms/spaces which we already have (though the differentiation for a space is very minor, different rounding of the avatar).
matrix.to doesn't ever have a Matrix token, so can only use unauthenticated APIs such as the (MSC) Room Preview API and Public Rooms (unfiltered) APIs. Neither give it access to message (permalink) previews. A new MSC would be needed or matrix.to would need to be registered/logged in per-user

**Comment by gsouquet** [comment]
@t3chguy You're right, the idea is not to holistically fix the mobile experience. We're going to solely focus on https://github.com/vector-im/element-web/issues/14962 and do a bit of CSS wrangling for a set period of time to try to tackle all the low hanging fruits to improve the readibility that we seem to have lost over time.
